### ISSUE TITLE
Anrufbeantworter: "Keine Verbindung" oder Stille beim Abnehmen

### ISSUE THREAD

**Issue by d4k0** [haumacher/phoneblock#101]
Letzte Woche habe ich den Anrufbeantworter in meiner FRITZ!Box 7590 AX ( FRITZ!OS: 8.00) eingerichtet, beim ersten Test über einen internen Anruf hat auch alles funktioniert (inklusive Ansagen). Aber schon etwas später am selben Tag klappte es nicht mehr, beim Anrufen des Anrufbeantworters kam nach einigen Sekunden nur die Meldung "Keine Verbindung" am Telefon (vorher zeigt das Telefon noch an, wie es die interne Nummer des AB anruft). Wenn ich dann den Anrufbeantworter auf der Webseite (mehrmals) aus- und wieder anschalte, klappt es (irgendwann) wieder.

Seit kurzem habe ich zusätzlich das Problem, dass der Anrufbeantworter zwar ran geht, es herrscht dann aber eine Stille (auch Sprechen bringt nichts). Der Anruf wird in diesem Fall jedoch auf der PhoneBlock-Oberfläche registriert.

Mir ist das erst heute wieder aufgefallen, da ich gestern Abend eine Rufumleitung für Kontakte aus dem lokalen PhoneBlock-Telefonbuch  an den Anrufbeantworter eingerichtet habe, heute Vormittag habe ich jedoch die Meldung erhalten, dass es einen Anruf in Abwesenheit aus diesem Telefonbuch gab.

An was kann das liegen? Die FRITZ!Box zeigt im Log keinerlei Fehler an, der Anrufbeantworter zeigt auch ständig "aktiv" an (selbst in den Fällen, in denen ich die Meldung "Keine Verbindung" erhalte).

Ich verwende MyFRITZ! als DynDNS-Provider. Der Anrufbeantworter ist ab-8340928260787xxx.

**Comment by KobeLAL** [comment]
Wenn ich Intern den AB Anrufe, höre ich auch nichts. Nach ein paar Sekunden kommt das Belegt Zeichen. Das gleiche hatte ich schonmal: https://github.com/haumacher/phoneblock/issues/81

Mir wird allerdings in der AB Phoneblock App kein Anruf signalisiert.

**Comment by d4k0** [comment]
Ich habe gestern noch einmal getestet und einen neuen AB (ab-8113874990237xxx) erstellt. In der FRITZ!Box habe ich auch alles noch einmal nach Anleitung neu eingerichtet .

Beim ersten und zweiten Test gegen 17:40 hatte ich wieder das Problem, dass der AB zwar ran geht, aber keine Ansage kommt (der Anruf wird auf der AB-Oberfläche registriert). Beim dritten Versuch kam dann am Telefon die Meldung "Keine Verbindung", der Anruf wurde auf der AB-Oberfläche nicht aufgezeichnet.

Kurz nach Mitternacht habe ich noch einmal einen Versuch gewagt, hier klappte es dann endlich zwei Mal in Folge mit der Ansage, die Anrufe wurden auch registriert. Aber wieder etwas später tauchte dann wieder das Problem auf, dass der AB zwar ran geht, aber nur Stille herrscht (Anruf wird auf der AB-Oberfläche registriert).

Heute funktionierte der AB bisher gar nicht, beim internen Anruf kommt nach ein paar Sekunden immer "Keine Verbindung". Wahrscheinlich müsste ich den AB wieder neu starten, damit dieser wieder ran geht.

Ich habe es heute auch einmal mit "FRITZ!App Fon" probiert, hier kommt dann in diesem Fall interessanterweise nach ein paar Sekunden die Meldung "Gegenstelle besetzt":

![FRITZ!App Fon](https://github.com/user-attachments/assets/fecc9cc8-b12a-49d5-adfc-717d678ead4c)

Könnte diese Meldung ein Hinweis auf das Problem geben? Erlaubt der AB z. B. nur ein bestimmte Anzahl an Anrufen?

**Es gibt also drei Fälle:**

1. Der AB nimmt an und die Ansagen werden abgespielt. Der Anruf wird auf der AB-Oberfläche registriert.
2. Der AB nimmt an, aber es herrscht nur Stille. Der Anruf wird auf der AB-Oberfläche registriert.
3. Beim Anrufen des ABs kommt nach einigen Sekunden die Meldung "Keine Verbindung" bzw. in der App "Gegenstelle besetzt", anschließend kommt das Besetzt-Zeichen bzw. es wird aufgelegt (in der App). Der Anruf wird auf der AB-Oberfläche NICHT registriert.

---

@haumacher 

> Der Anruf "in Abwesenheit" - wann genau war der - vielleicht kann ich etwas in den Log-Dateien nachvollziehen. Obwohl ich auf dem Live-System nur ganz wenig Logging aktiv habe - aber wer weiß.

Der war um 8:28 am 19.11.2024. Aber das Problem scheint ja irgendwie zu sein, dass die Verbindung zum AB nach kurzer Zeit irgendwie "weg" ist bzw. der AB "besetzt" ist (laut App).

Gibt es in der FRITZ!Box neben System > Ereignisse noch ein ausführlicheres Log, in dem mehr zu finden sein könnte? Oder gibt es vielleicht einen Test-Server (ich glaube, ich habe das in einem anderen Ticket gelesen), wo mehr geloggt wird?

Ich werde es morgen noch einmal mit dem Phoneblock-DynDNS testen, auch wenn das eigentlich nicht das Problem sein sollte, da MyFRITZ! die aktuelle IP liefert (die IP hat sich zudem eh nicht geändert).

---

@KobeLAL 

Das hört sich schon wie mein Problem an, in meiner Liste wäre das Fall 3. Hilft bei dir das Aus- und Einschalten des ABs (dazwischen vielleicht ein paar Sekunden warten)? Ich komme dann zumindest immer/meist zu Fall 2.

**Comment by haumacher** [comment]
@d4k0 Danke für den Bericht. Der AB hat keine Beschränkung bzgl. der Anzahl von Anrufen. Wer will, kann sich den ganzen Tag mit ihm unterhalten. Ich habe bisher noch keine Idee, was das Problem ist. Ich melde mich, wenn ich dazukomme das zu untersuchen.

**Comment by geimist** [comment]
> Gibt es in der FRITZ!Box neben System > Ereignisse noch ein ausführlicheres Log, in dem mehr zu finden sein könnte?

Unter http://fritz.box/support.lua kannst du dir Supportdaten erstellen. Dort sind auch Details zu SIP-Verbindungen enthalten.    

<img width="210" alt="Bildschirmfoto 2024-11-22 um 11 05 20" src="https://github.com/user-attachments/assets/e9e81f4a-6838-4916-a972-e50232ae117a">

**Comment by KobeLAL** [comment]
Kurz den AB auf der Phoneblock Website aus und angeschaltet. Jetzt kommt nach ein paar Sekunden kein belegt Zeichen. Man hört nur Stille, sprich keine Audio. Auf der Phoneblock Website wird der Anruf registriert. Irgendwo ist der Wurm drin.

**Comment by d4k0** [comment]
Ich habe gestern beim Testen eine interessante Entdeckung gemacht: Beim Fall 2, wenn der AB zwar ran geht, aber es nur Stille herrscht, habe ich aus Spaß einfach einmal gewartet, um zu sehen, ob noch etwas passiert. Und tatsächlich fängt dann die Ansage _irgendwann_ an. Einmal waren es vielleicht 20 Sekunden, dann über eine Minute und ein anderes Mal über zwei Minuten. Das scheint aber irgendwie willkürlich zu sein.

@KobeLAL 
Kannst du dieses Verhalten vielleicht bestätigen?

---

Seit gestern ist aber bisher das Problem, dass der AB nicht erreicht werden kann und ein Besetzt-Zeichen kommt ("Keine Verbindung"), noch nicht wieder aufgetaucht. Daher konnte ich die Logs dazu nicht auswerten. Ich habe das aber weiter auf dem Schirm.

@geimist 
Danke für den Hinweis auf die Stelle/URL, ich wusste doch, dass ich das schon früher Mal gemacht habe, um einen Fehler zu finden.

---

@haumacher 

> Der AB hat keine Beschränkung bzgl. der Anzahl von Anrufen.

Das habe ich schon vermutet, dann sollte es nicht daran liegen. Ich dachte vielleicht, dass es z. B. einen Schutz vor zu vielen Anfragen von einer IP/Telefonnummer gibt.

Ansonsten wüsste ich gerade auch nicht, was bei mir "anders" sein könnte, so dass es nicht (immer) funktioniert. In der FRITZ!Box habe ich nur andere DNS-Server eingerichtet, das sollte aber auch nicht das Problem sein. Vielleicht gibt das FRITZ!Box-Log einen Hinweis, wenn der AB gar nicht erreicht werden kann.

**Comment by KobeLAL** [comment]
@d4k0 : Ja, das kann ich bestätigen. Nach ca. 2 Min 20 Sek. fängt er plötzlich an zu sprechen.

**Comment by KobeLAL** [comment]
Weiß man mittlerweile an was es liegt? Bei mir tritt es leider weiterhin auf. Man merkt es, wenn in der Anrufliste ein Anruf über SPAM abgefangen wurde, das der Anrufbeantworter wieder nicht erreichbar ist. Wenn man Intern anruft, kommt das besetzt Zeichen.

**Comment by haumacher** [comment]
Um das zu analyiseren, benötige ich die ID des Anrufbeantworters und den genauen Zeitpunkt des Testanrufs. Dann kann ich zumindest Feststellen, ob bei PhoneBlock eine Nachricht eingetroffen ist.

**Comment by KobeLAL** [comment]
Die ID ist ab-35177875584XXXXX. Den genauen Zeitpunkt habe ich jetzt leider nicht mehr. Probieren kann ich es jetzt zwar nochmal, aber ich hab auf der AB-Seite den AB kurz deaktiviert und wieder aktiviert. Jetzt nimmt er wieder ab, aber eben ohne Audiodatei bzw. die spielt sich nach ein paar Minuten (siehe oben) dann doch ab. Eben 10:08 den Ab Intern angerufen und er hat gleich abgenommen und die Audio abgespielt.

**Comment by KobeLAL** [comment]
Gerade 9:22 Uhr Intern angerufen. Die Audio wird nicht abgespielt. Wahrscheinlich wie oben, erst später. Werde mich melden, wenn wieder das besetzt Zeichen kommt.

**Comment by KobeLAL** [comment]
Richtig, solange der Anruf entgegen genommen wird, ist alles ok. Problem sehe ich eher, warum nach ein paar Tagen/Wochen nur noch das besetzt Zeichen kommt. Da werden dann die Anrufe nicht mehr abgefangen.

**Comment by haumacher** [comment]
Das ist richtig, aber das war heute um 9:22 nicht der Fall?

**Comment by KobeLAL** [comment]
Nein, heute nicht. Ich melde mich wieder wenn das besetzt Zeichen wieder da ist.

**Comment by geimist** [comment]
Ich hatte es jetzt auch mehrmals über mehrere Anschlüsse vergeblich versucht. 
Gerade eben:
`ab-102916529890xxxx` 15:30 Uhr ➜ "_keine Verbindung_" (Besetztzeichen)

**Comment by KobeLAL** [comment]
Weißt du was neues, warum das passiert @haumacher?

**Comment by haumacher** [comment]
> Hallo zusammen, ich beobachte das selbe Verhalten. Eine Ansage habe ich nie erhalten. Immer Stille oder (meistens) besetzt.
> 
> ab-7160127517023178
> 
> Testanrufe (MEZ)
> 
>     * 23.12.24, 10:48: besetzt
> 
>     * 23.12.24, 11:00 besetzt
> 
> 
> <AB deaktiviert, aktiviert>
> 
>     * 23.12.24, 11:05: Stille, Anrufdauer 3min, Anruf registriert
> 
>     * 23.12.24, 11:10: besetzt

Die Log-Einträge hierzu sehen absolut unverdächtig aus:

```
phoneblock.log:[2024-12-23 08:44:40] INFO: [de.haumacher.phoneblock.ab.CreateABServlet]: Created answerbot 'ab-716012xxx' for: ddd35eab-0ea7-xxx
phoneblock.log:[2024-12-23 08:52:35] INFO: [de.haumacher.phoneblock.ab.SipService]: No active registration for user 'ab-716012xxx'.
phoneblock.log:[2024-12-23 08:52:35] INFO: [de.haumacher.phoneblock.ab.SipService]: Resolved IPv6 address for 'ab-716012xxx' (hjoxxx.myfritz.net): 2003:de:afff:xxx
phoneblock.log:[2024-12-23 08:52:35] INFO: [de.haumacher.phoneblock.ab.SipService]: Started registration for ab-716012xxx.
phoneblock.log:[2024-12-23 08:52:35] INFO: [de.haumacher.phoneblock.ab.SipService]: Sucessfully registered ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:02:38] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:06:44] INFO: [de.haumacher.phoneblock.answerbot.AnswerBot]: Completed SPAM call for 'ab-716012xxx' with 'ab-**5*xxx' started Mon Dec 23 09:05:44 UTC 2024 talked for 59.91 seconds.
phoneblock.log:[2024-12-23 09:08:05] INFO: [de.haumacher.phoneblock.answerbot.AnswerBot]: Completed SPAM call for 'ab-716012xxx' with 'ab-**5*xxx' started Mon Dec 23 09:07:15 UTC 2024 talked for 50.081 seconds.
phoneblock.log:[2024-12-23 09:16:21] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:28:11] INFO: [de.haumacher.phoneblock.ab.SipService]: Stopping answer bot 'ab-716012xxx'.
phoneblock.log:[2024-12-23 09:28:11] INFO: [de.haumacher.phoneblock.ab.SipService]: No active registration for user 'ab-716012xxx'.
phoneblock.log:[2024-12-23 09:28:11] INFO: [de.haumacher.phoneblock.ab.SipService]: Resolved IPv6 address for 'ab-716012xxx' (hjoxxx.myfritz.net): 2003:de:afff:xxx
phoneblock.log:[2024-12-23 09:28:11] INFO: [de.haumacher.phoneblock.ab.SipService]: Started registration for ab-716012xxx.
phoneblock.log:[2024-12-23 09:28:12] INFO: [de.haumacher.phoneblock.ab.SipService]: Sucessfully registered ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:35:31] INFO: [de.haumacher.phoneblock.answerbot.AnswerBot]: Completed SPAM call for 'ab-716012xxx' with 'ab-**5*xxx' started Mon Dec 23 09:28:17 UTC 2024 talked for 433.858 seconds.
phoneblock.log:[2024-12-23 09:37:53] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:42:53] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 09:50:13] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 10:02:10] INFO: [de.haumacher.phoneblock.ab.SipService]: Stopping answer bot 'ab-716012xxx'.
phoneblock.log:[2024-12-23 10:02:16] INFO: [de.haumacher.phoneblock.ab.SipService]: No active registration for user 'ab-716012xxx'.
phoneblock.log:[2024-12-23 10:02:16] INFO: [de.haumacher.phoneblock.ab.SipService]: Resolved IPv6 address for 'ab-716012xxx' (hjoxxx.myfritz.net): 2003:de:afff:xxx
phoneblock.log:[2024-12-23 10:02:16] INFO: [de.haumacher.phoneblock.ab.SipService]: Started registration for ab-716012xxx.
phoneblock.log:[2024-12-23 10:02:16] INFO: [de.haumacher.phoneblock.ab.SipService]: Sucessfully registered ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 10:08:12] INFO: [de.haumacher.phoneblock.answerbot.AnswerBot]: Completed SPAM call for 'ab-716012xxx' with 'ab-**5*xxx' started Mon Dec 23 10:05:06 UTC 2024 talked for 185.89 seconds.
phoneblock.log:[2024-12-23 10:33:05] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 11:21:13] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
phoneblock.log:[2024-12-23 12:21:38] INFO: [de.haumacher.phoneblock.ab.SipService]: Updated registration ab-716012xxx: 200 OK
```

Der AB war registriert und er hat auch gesprochen. Ich kann mir das eigentlich nur so erklären, dass es sich um ein Netzwerkproblem handelt - es scheint so, als ob die RTP Sprachpakete nicht durchkommen wollen. Bei meinem eigenen Anschluss funktioniert das allerdings die letzte Zeit problemlos.

Mit dem nächsten Update wird man den AB auf IPv4 umstellen können - vielleicht kann man damit nochmal testen, ob das irgendwie einen Unterschied ergibt - aktuell bin ich aber einigermaßen ratlos.

**Comment by Obihoernchen80** [comment]
@haumacher 
Der Anruf der durchging mag normal aussehen, aber das deutlich größere Problem ist bei mir dass quasi immer besetzt ist.

**Comment by geimist** [comment]
Ich für meinen Fall habe festgestellt, dass ich bei der ganzen Testerei den Online-AB deaktiviert hatte. Ich hab das erst mitbekommen, als ich im Sip-Log den Fehler `SIP/2.0 401 Unauthorized` untersuchte.

@Obihoernchen80 Aktiviert ist dein AB im Webinterface? Ich vermute ja, wenn der Anruf auch geloggt wird.

**Comment by Obihoernchen80** [comment]
> @Obihoernchen80 Aktiviert ist dein AB im Webinterface?

Soweit ich das verstehe, ja.

**Comment by haumacher** [comment]
Ich glaube, ich habe ein Problem identifiziert und behoben: Es wurden teilweise dringende Aufgaben - wie z.B. das Aufsetzen von Threads für die Sprachübertragung verzögert, weil der Thread-Pool ausgeschöpft war. Wenn so etwas passiert, dann kann es sein, dass der Anruf zwar angenommen wird, aber kein Ton übertragen wird.

Bitte das Thema weiter beobachten, bzw. nochmal testen.

**Comment by KobeLAL** [comment]
Werde es mal beobachten. Hoffe das es der Fehler war und auch das weg ist, das nur noch das Besetzt Zeichen kommt.
